apis/clusterresources/v1beta1/opensearchegressrules_types.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type OpenSearchEgressRulesSpec struct {
+	ClusterID           string `json:"clusterId"`
+	Name                string `json:"name"`
```

Comment:
As Instaclustr API [doc](https://instaclustr.redoc.ly/Current/tag/OpenSearch-Egress-Rules-V2#paths/~1cluster-management~1v2~1resources~1applications~1opensearch~1egress-rules~1v2/post) says the field `name` should be returned by the API if you add a new egress rule, but this is not part of a creation request. You should move this field from the resource spec to its status

Reply:
I've looked at the request sample in the doc, and the `name` field` is passed there. I think you should test what if you don't pass it. If any error is returned then change nothing

Reply:
It`s described in spec file, and in request body, in general this field does not affect nothing, but I think we should implement code according to spec